quill-ndbc/src/main/scala/io/getquill/context/ndbc/BaseNdbcContext.scala

```diff
@@ -0,0 +1,106 @@
+package io.getquill.context.ndbc
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+import org.slf4j.LoggerFactory
+
+import com.typesafe.scalalogging.Logger
+
+import io.getquill.NamingStrategy
+import io.getquill.context.sql.SqlContext
+import io.getquill.context.sql.idiom.SqlIdiom
+import io.trane.future.scala._
+import io.trane.ndbc.DataSource
+import io.trane.ndbc.Row
+import java.time.ZoneOffset
+import io.trane.ndbc.PreparedStatement
+
+abstract class BaseNdbcContext[Dialect <: SqlIdiom, Naming <: NamingStrategy, P <: PreparedStatement, R <: Row](
```

Comment:
How about we use `ContextEffect` here instead of `JFuture.toScala`? We can have a `NdbcContextEffect` subclass that will just delegate to `JFuture.toScala` but we can just as easily implement a `NdbcMonixContextEffect` which will wrap everything in Monix `Task`s. Then we get a `NdbcMonixContext` for free!